Source/Core/DolphinQt/HotkeyScheduler.cpp

```diff
@@ -139,10 +139,22 @@ void HotkeyScheduler::Run()
 
   while (!m_stop_requested.IsSet())
   {
-    Common::SleepCurrentThread(1000 / 60);
+    const u64 freelook_update_freq = 200;
+    const u64 interval = SystemTimers::GetTicksPerSecond() / freelook_update_freq;
+    const u64 now = CoreTiming::GetTicks();
 
-    if (Core::GetState() == Core::State::Uninitialized || Core::GetState() == Core::State::Paused)
+    if (now - m_last_freelook_ticked > interval)
+    {
+      m_last_freelook_ticked = now;
+      g_controller_interface.SetCurrentInputChannel(ciface::InputChannel::FreeLook);
       g_controller_interface.UpdateInput();
+      FreeLook::UpdateInput();
+    }
+
+    Common::SleepCurrentThread(1000 / 60);
```

Comment:
If this sleeps the thread for ~16ms, how is freelook going to update at 200Hz? if (now - m_last_freelook_ticked > interval) is always gonna be true.

Reply:
This is a good point.  I thought I had tested that and this way felt closer to master but it does seem odd now that you mention it.

Reply:
I guess you had not noticed because games don't run above 60 unless the speed is unlocked, but with a sleep in the middle, this is unlikely to actually update at 60, it's gonna lose time and you are going to miss input updates on frames quite often.

Reply:
If it works like I'm guessing, freelook input should be updated just before rendering starts, or as late as possible within a frame anyway, and should be in the cpu thread. Otherwise it's never gonna be accurate.

Reply:
Well, from that standpoint it should exist on the graphics thread (which may or may not be the cpu thread if dual core is enabled).  However, I'm not sure if tying inputs to framerate is a great idea.

Reply:
Sure, why not? If I run the game at 200%, freelook should update inputs at twice the speed.
And you tick timing checks seem to be exactly for that reason.

Reply:
After the edit to your comment: you seem to want to link Freelook updates to the game speed, as that's what you'd get if you check the emulation ticks.
If you are increasing values by a fixed amount every frame, that is wrong and you should simply use a delta time there.